src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -171,7 +170,12 @@ export function initDashboardApp(app, deps) {
             history.push(deps.addBasePath(viewUrl));
           };
           $scope.delete = (dashboards) => {
-            return service.delete(dashboards.map((d) => d.id));
+            const ids = dashboards.map((d) => ({ id: d.id, appId: d.appId }));
+            return Promise.all(
+              ids.map(({ id, appId }) => {
+                return deps.savedObjectsClient.delete(appId, id);
+              })
+            );
```

Comment:
@pjfitzgibbons Can you explain this fix a bit more? Is there a reason why you switched from using the `savedDashbaoards` service to using the `savedObjectsClient` directly? It seems like the old way would potentially be preferable if it actually is a bulk action under the hood rather than multiple calls with client for each object (although I don't remember enough about the implementation under the hood to know if it's actually a bulk operation or whether it does a similar loop).

Reply:
Unless You Know Better :) .... there is no API on SavedObject that I can bulk-delete a collection of {id:, type:}, since the dashboard-list _could_ contain multiple plugin's items, ... different "type"s... so the SavedObjectSevice is no longer an adequate API for this.
The alternative, and workaround for 2.7, is to iterate the selected list and delete.   

This *should* be a rarely-used API, and the most any user could select is all 50 (100?) listed on a single page, so even that is not horrible abuse of the API (might take a minute to complete).

Thoughts? @joshuarrrr @abbyhu2000

Reply:
Yeah - confirmed this is what we do in `visualize`: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/685c9115f4779e2743acbd3d2ff17e6c3d4ee6b3/src/plugins/visualize/public/application/components/visualize_listing.tsx#L135-L148